src/partition_mesh.py

```diff
@@ -376,27 +385,21 @@ def write_meshes(meshes, recoveryInfo, meshname: str, orig_mesh) -> None:
     """
     Writes meshes to given directory.
     """
-    # Get the absolute directory where we want to store the mesh
-    dirname = os.path.abspath(meshname)
     # Strip off the mesh-prefix for the directory creation
     mesh_prefix=os.path.basename(os.path.normpath(meshname))
-    logging.info(dirname)
-    recoveryName = os.path.join(dirname, 'recovery.json')
-    if os.path.exists(dirname):
-        shutil.rmtree(dirname)
-    os.mkdir(dirname)
+    recoveryName = 'recovery.json'
     for i in range(len(meshes)):
         mesh = meshes[i]
-        write_mesh(dirname + "/" + mesh_prefix + "_" + str(i) + ".vtu", mesh.points, mesh.data_index, mesh.cells,
+        write_mesh(mesh_prefix + "_" + str(i) + ".vtu", mesh.points, mesh.data_index, mesh.cells,
```

Comment:
Due to the last changed I switched the name here to be only the `mesh_prefix` (`path.basename`). This won't work if I try to store the mesh in a directory now, say `-m partitioned/result`, the result will be stores in `result` only.

Reply:
Ah, I missed this point. Can't we simply directly remove  line `389` 
```python
# Strip off the mesh-prefix for the directory creation
 mesh_prefix=os.path.basename(os.path.normpath(meshname))
```

The creation of directory is responsibility of the user. We do not need to strip mesh name anymore?

Reply:
Yes, it's the responsibility of the user to take select a proper location. We can still help the user by going with either of the above options, i.e., the user specifies a directory -> create one

Reply:
I would prefer using another argument `--directory -dir` for specifying the directory, it is more explicit and clear than giving the directory with the mesh prefix.

Reply:
You can do this as long as the behavior is for all scripts the same, e.g. `join_mesh` also generates output meshes.